controllers/cluster_upgrade.go

```diff
@@ -770,6 +771,7 @@ func (r *ClusterReconciler) upgradeInstanceManager(
 // upgradeInstanceManagerOnPod upgrades an instance manager of a Pod via an HTTP PUT request.
 func upgradeInstanceManagerOnPod(
 	ctx context.Context,
+	statusClient *instance.StatusClient,
```

Comment:
I think this needs to be a different client. The timeout we use for the status client could hurt here.

Reply:
This is solved by configuring the timeout before executing the request.